xmr-btc/src/bitcoin.rs

```diff
@@ -212,6 +219,11 @@ pub trait GetRawTransaction {
     async fn get_raw_transaction(&self, txid: Txid) -> Result<Transaction>;
 }
 
+#[async_trait]
+pub trait Network {
```

Comment:
@rishflab @D4nte Like all the other traits that the wallet must fulfill I added a `Network` trait so the wallet is initialized for a network. But at the moment there is no validation for this. 
Additionally, we added `Config` that allows specifying a network so the boundaries when to act depend on the network. Should these concepts be combined?

Reply:
I guess they should be combined yes, is `Wallet` the best place to store the network?

Reply:
The wallet still has to be initialized with `Network`, because the `xmr_btc` crate requires the network (when constructing `TxLock` - only the wallet is available). 

for now I added the `bitcoin_network` to the `Config` and always just pass that network (always use the `Config`. I think that is fine for now, eventually we can come up with a better solution.